packages/esm-patient-search-app/src/patient-search/patient-search.component.tsx

```diff
@@ -1,98 +1,88 @@
 import React, { useEffect } from 'react';
 import { useTranslation } from 'react-i18next';
-import isEmpty from 'lodash-es/isEmpty';
 import { usePagination } from '@openmrs/esm-framework';
 import { Loading, PaginationNav, Tile } from 'carbon-components-react';
-import { SearchedPatient } from '../types/index';
 import EmptyDataIllustration from './empty-data-illustration.component';
 import PatientSearchResults from '../patient-search-result/patient-search-result.component';
 import styles from './patient-search.scss';
+import { usePatients } from '../hooks/usePatients';
 
 const resultsPerPage = 5;
 
 interface PatientSearchProps {
   hidePanel?: () => void;
-  searchResults: Array<SearchedPatient>;
-  status: 'searching' | 'resolved' | 'error' | 'idle';
-  error;
+  querySearchTerm: string;
 }
 
-const PatientSearch: React.FC<PatientSearchProps> = ({ hidePanel, searchResults, status, error }) => {
+const PatientSearch: React.FC<PatientSearchProps> = ({ hidePanel, querySearchTerm }) => {
   const { t } = useTranslation();
-  const { totalPages, currentPage, goToNext, goToPrevious, results, goTo } = usePagination(
-    searchResults,
-    resultsPerPage,
-  );
-
+  const { patients, isLoading, error } = usePatients(querySearchTerm);
+  const { totalPages, results, goTo } = usePagination(patients, resultsPerPage);
   const handlePageChange = (page: number) => {
-    if (page === 0 && currentPage === 0) {
-      goToNext();
-    } else if (page + 1 > currentPage) {
-      goToNext();
-    } else if (page + 1 < currentPage) {
-      goToPrevious();
-    }
+    goTo(page + 1);
   };
 
   useEffect(() => {
-    if (searchResults.length) {
-      goTo(0);
+    if (patients.length) {
+      goTo(1);
     }
-  }, [searchResults]);
+  }, [patients.length, querySearchTerm]);
 
+  if (error) {
+    return (
+      <div className={styles.searchResults}>
+        <p className={styles.resultsText}>{t('errorText', 'An error occurred while performing search')}</p>
+        <Tile className={styles.emptySearchResultsTile}>
+          <EmptyDataIllustration />
```

Comment:
Probably time to start thinking of ways to factor out these error tiles. CC @denniskigen @ZacButko